tests/Dialect/TensorExt/IR/convert_layout_folder.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: heir-opt --canonicalize --split-input-file %s | FileCheck %s
+
+#map = affine_map<(d0) -> (d0 * 32)>
+#map1 = affine_map<(d0) -> (d0)>
+
+// CHECK-LABEL: func @fold
+// CHECK-NOT: tensor_ext.convert_layout
+// CHECK: return
+func.func @fold(%arg0: !secret.secret<tensor<32xi16>> {tensor_ext.layout = #map1}) -> (!secret.secret<tensor<32xi16>> {tensor_ext.layout = #map}) {
+  %0 = secret.generic ins(%arg0 : !secret.secret<tensor<32xi16>>) attrs = {arg0 = {layout = #map}, layout = [#map]} {
```

Comment:
You can probably remove the secret stuff from this test, just use plain tensors for testing.